polar-core/src/terms.rs

```diff
@@ -43,6 +43,7 @@ pub struct ExternalInstance {
     pub instance_id: u64,
     pub constructor: Option<Term>,
     pub repr: Option<String>,
+    pub class_repr: Option<String>,
```

Comment:
Is there a reason not to have this reference the instance ID of the class instance itself, instead of a string representation? I feel like that is a direction we've talked about moving before (i.e. it would make it trivial to prevent `isa` checks from crossing the FFI boundary), and it feels like it'd be essentially this same change with only some minor tweaks.

Reply:
the reason for choosing `Option<String>` over `class_id` was one of implementation expediency. I originally started with that plan but chose the string alternative as it could be implemented without necessitating FFI changes in all of the host languages which turned out to be somewhat tricker than I expected.

Reply:
We're going to make the name -> ID shift in a follow-up